### PR TITLE
[#4004] fixed firstPull address choose logic

### PR DESCRIPTION
**Pull request by chengyouling**

### CODE REVIEW

File: clients/config-kie-client/src/main/java/org/apache/servicecomb/config/kie/client/KieConfigManager.java [hunk 1]
```diff
@@ -50,24 +51,27 @@ public class KieConfigManager extends AbstractTask {
...
   }
 
   public void firstPull() {
     try {
       Map<String, Object> data = new HashMap<>();
+      String address = kieAddressManager.chooseFirstPullAddress();
       this.configurationsRequests.forEach(r -> {
         r.setRevision(ConfigurationsRequest.INITIAL_REVISION);
-        ConfigurationsResponse response = configKieClient.queryConfigurations(r);
+        ConfigurationsResponse response = configKieClient.queryConfigurations(r,address);
```
Comment (liubao68) [inline clients/config-kie-client/src/main/java/org/apache/servicecomb/config/kie/client/KieConfigManager.java:74]: Format

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/cc/ConfigCenterDynamicPropertiesSource.java [hunk 2]
```diff
@@ -80,21 +80,21 @@ private void init(Environment environment) {
         createConfigCenterConfiguration(configCenterConfig);
 
     QueryConfigurationsRequest queryConfigurationsRequest = firstPull(configCenterConfig, configCenterClient,
-        environment);
+        environment, kieAddressManager);
```
Comment (liubao68) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/cc/ConfigCenterDynamicPropertiesSource.java:83]: rename

File: clients/config-kie-client/src/main/java/org/apache/servicecomb/config/kie/client/KieConfigManager.java (hunk 1 above)
Comment (chengyouling) [reply clients/config-kie-client/src/main/java/org/apache/servicecomb/config/kie/client/KieConfigManager.java]: fixed

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/cc/ConfigCenterDynamicPropertiesSource.java (hunk 2 above)
Comment (chengyouling) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/cc/ConfigCenterDynamicPropertiesSource.java]: fixed

File: clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java [hunk 3]
```diff
@@ -148,10 +151,21 @@ public String formatUrl(String url, boolean absoluteUrl, String address) {
...
-      return getDefaultAddress();
+      address = getDefaultAddress();
+    } else {
+      address = getAvailableZoneAddress();
+    }
+    if (telnetTest(address)) {
+      return address;
     } else {
-      return getAvailableZoneAddress();
+      if (addressRetry.getAndIncrement() >= 3) {
```
Comment (liubao68) [inline clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java:163]: This is not correct. And it is better to add a tests case for choose address.

File: clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java [hunk 4]
```diff
@@ -148,10 +151,21 @@ public String formatUrl(String url, boolean absoluteUrl, String address) {
...
+        LOGGER.warn("choose address has retry three times, no availible address, return current address {}", address);
```
Comment (liubao68) [inline clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java:164]: telnet will give errors. this method may print too many logs.

File: clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java (hunk 3 above)
Comment (chengyouling) [reply clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java]: fixed

File: clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java (hunk 4 above)
Comment (chengyouling) [reply clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java]: fixed

**Review by liubao68 — APPROVED** [review]

**Pull request merged**